### ISSUE TITLE
New Feature: Metastore Database Prefix

### ISSUE THREAD

**Issue by zshao** [airbnb/reair#44]
Allow a database prefix to be added to the destination cluster.  This can be used for production testing .  For example, add "test_" to all database names on the destination cluster.

**Comment by jingweilu1974** [comment]
Yeah, sounds good to me.

**Comment by plypaul** [comment]
I agree, it'd be useful. The last time I looked into this though, it was a bit more complicated than expected. One approach might be to create another implementation of `DestinationObjectFactory` that creates objects with the modified prefix. However, you would need to make sure that all path generation goes through that object. Some effort was made to cover the cases, but I don't think all of them were addressed.

**Comment by ericsahit** [comment]
sounds useful to me
